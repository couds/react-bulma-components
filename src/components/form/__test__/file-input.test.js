import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { JSDOM } from 'jsdom';

import { InputFile } from '..';

describe('Dropdown component', () => {
  beforeEach(() => {
    // eslint-disable-next-line
    global.window = new JSDOM('<body><div id="app-root"></div></body>').window;
  });
  it('Should Exist', () => {
    expect(InputFile).toMatchSnapshot();
  });
  it('Should render file input tree', () => {
    const component = renderer.create(<InputFile onChange={() => {}} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should call onChange with file data', () => {
    const onChange = jest.fn();
    const component = shallow(
      <InputFile
        className="input-file"
        value=""
        style={{ width: 400 }}
        onChange={onChange}
      />,
    );
    component.find('.input-file .file-input').simulate('change', {
      target: {
        files: [
          {
            name: 'dummyValue.something',
          },
        ],
      },
    });
    expect(component.state('filename')).toBe('dummyValue.something');
    expect(onChange).toBeCalled();
  });

  it('should call onChange without file data', () => {
    const onChange = jest.fn();
    const component = shallow(
      <InputFile
        className="input-file"
        value=""
        style={{ width: 400 }}
        onChange={onChange}
      />,
    );
    component.find('.input-file .file-input').simulate('change', {
      target: {
        files: [],
      },
    });
    expect(component.state('filename')).toBe(undefined);
    expect(onChange).toBeCalled();
  });
  it('should pass file attributes', () => {
    const inputProps = {
      accept: 'image/*',
      capture: 'source',
      multiple: true,
    };
    const component = renderer.create(
      <InputFile inputProps={inputProps} onChange={() => {}} />,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
