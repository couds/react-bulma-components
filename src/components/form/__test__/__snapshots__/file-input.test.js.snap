// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown component Should Exist 1`] = `[Function]`;

exports[`Dropdown component Should render file input tree 1`] = `
<div
  className="file"
>
  <label
    className="file-label"
  >
    <input
      className="file-input"
      onChange={[Function]}
      type="file"
    />
    <span
      className="file-cta"
    >
      <span
        className="file-label"
      >
        Choose a file...
      </span>
    </span>
  </label>
</div>
`;

exports[`Dropdown component should pass file attributes 1`] = `
<div
  className="file"
>
  <label
    className="file-label"
  >
    <input
      accept="image/*"
      capture="source"
      className="file-input"
      multiple={true}
      onChange={[Function]}
      type="file"
    />
    <span
      className="file-cta"
    >
      <span
        className="file-label"
      >
        Choose a file...
      </span>
    </span>
  </label>
</div>
`;
